core/sail/memory/src/main/java/org/eclipse/rdf4j/sail/memory/model/MemIRI.java

```diff
@@ -130,6 +130,19 @@ public int hashCode() {
 		return hashCode;
 	}
 
+	@Override
+	public int compareTo(IRI o) {
+		if (o instanceof MemIRI) {
+			int compareTo = namespace.compareTo(((MemIRI) o).namespace);
+			if (compareTo == 0) {
+				return localName.compareTo(((MemIRI) o).localName);
+			} else
+				return compareTo;
+		}
+
+		return toString().compareTo(o.toString());
+	}
+
```

Comment:
This means we don't have to instantiate new string objects (concat) when comparing two MemIRIs. Biggest difference to be honest.